transports/bifrost-http/lib/config.go

```diff
@@ -1737,28 +1742,30 @@ func (c *Config) EditMCPClientTools(ctx context.Context, name string, toolsToExe
 		return fmt.Errorf("no MCP config found")
 	}
 
-	if err := c.client.EditMCPClientTools(name, toolsToExecute); err != nil {
-		return fmt.Errorf("failed to edit MCP client tools: %w", err)
-	}
-
-	// Find the client config and create an updated version
-	var updatedClientConfig schemas.MCPClientConfig
+	// Find the client config
 	var found bool
 	for i, clientConfig := range c.MCPConfig.ClientConfigs {
-		if clientConfig.Name == name {
-			c.MCPConfig.ClientConfigs[i].ToolsToExecute = toolsToExecute
-			updatedClientConfig = c.MCPConfig.ClientConfigs[i]
+		if clientConfig.ID == id {
+			// Update the in-memory config with new values
+			c.MCPConfig.ClientConfigs[i].Name = updatedConfig.Name
+			c.MCPConfig.ClientConfigs[i].Headers = updatedConfig.Headers
```

Comment:
you cannot directly update the headers, you need to check for env vars and substitute then with news opens in env before giving to bifrost client

- Check for new env vars, add their entry in env vars tables
- Remove old replaced entries
- give un redacted config to bifrost 

check similar logic for in memory provider/key updates